packages/core/src/lib/loaduikits.js

```diff
@@ -5,91 +5,110 @@ const _ = require('lodash');
 
 const logger = require('./log');
 
-let findModules = require('./findModules'); //eslint-disable-line prefer-const
-let fs = require('fs-extra'); // eslint-disable-line
-
-const uiKitMatcher = /^uikit-(.*)$/;
-const nodeModulesPath = path.join(process.cwd(), 'node_modules');
-
-/**
- * Given a path: return the uikit name if the path points to a valid uikit
- * module directory, or false if it doesn't.
- * @param filePath
- * @returns UIKit name if exists or FALSE
- */
-const isUIKitModule = filePath => {
-  const baseName = path.basename(filePath);
-  const engineMatch = baseName.match(uiKitMatcher);
+const { resolvePackageFolder } = require('./resolver');
 
-  if (engineMatch) {
-    return engineMatch[1];
-  }
-  return false;
-};
+let fs = require('fs-extra'); // eslint-disable-line
 
-const readModuleFile = (kit, subPath) => {
+const readModuleFile = (uikitLocation, subPath) => {
   return fs.readFileSync(
-    path.resolve(path.join(kit.modulePath, subPath)),
+    path.resolve(path.join(uikitLocation, subPath)),
     'utf8'
   );
 };
 
 /**
  * Loads uikits, connecting configuration and installed modules
- * [1] Looks in node_modules for uikits.
- * [2] Filter out our uikit-polyfills package.
- * [3] Only continue if uikit is enabled in patternlab-config.json
+ * [1] Lists the enabled uikits from patternlab-config.json
+ * [2] Try to resolve the location of the uikit in the package dependencies
+ * [3] Warn when the uikit couldn't be loaded
  * [4] Reads files from uikit that apply to every template
  * @param {object} patternlab
  */
 module.exports = patternlab => {
   const paths = patternlab.config.paths;
 
-  const uikits = findModules(nodeModulesPath, isUIKitModule) // [1]
-    .filter(kit => kit.name !== 'polyfills'); // [2]
-  uikits.forEach(kit => {
-    const configEntry = _.find(_.filter(patternlab.config.uikits, 'enabled'), {
-      name: `uikit-${kit.name}`,
-    }); // [3]
-
-    if (!configEntry) {
-      logger.warning(
-        `Could not find uikit with name uikit-${kit.name} defined within patternlab-config.json, or it is not enabled.`
-      );
-      return;
+  const uikitConfigs = _.filter(patternlab.config.uikits, 'enabled'); // [1]
+  uikitConfigs.forEach(uikitConfig => {
+    let uikitLocation = null;
+    if ('package' in uikitConfig) {
+      try {
+        uikitLocation = resolvePackageFolder(uikitConfig.package);
+      } catch (ex) {
+        logger.warning(
+          `Could not find uikit with package name ${uikitConfig.package}. Did you add it to the 'dependencies' section in your 'package.json' file?`
+        );
+        return;
+      }
```

Comment:
Please provide an example `patternlab-config.json` for that or change the given ones. The user shouldn't get warnings for a fresh installed pattern-lab after we merge this.

Reply:
`packages/core/test/util/patternlab-config.json` is left as is intentionally because it is used in the tests for checking the warning messages.

Question back: `packages/development-edition-engine-react/patternlab-config.json` doesn't have a `uikits` block, but does point to `./node_modules/@pattern-lab/uikit-workshop` files inside the `paths` section. With my change, this won't work in Yarn v2. I don't know if this is anything specific to React but I would need some help here on what to do.

Reply:
Leave it out then. I'm not shure which engines are used anymore. Some feel depreacated / outdated anyway.